lib/inspec/resources/sys_info.rb

```diff
@@ -84,6 +86,36 @@ def mac_hostname(opt = nil)
       end
     end
 
+    def solaris_hostname(opt = nil)
+      if opt
+        opt = case opt
+              when "f", "long", "fqdn", "full"
+                " -f"
+              when "d", "domain"
+                " -d"
+              when "i", "ip_address"
+                " -I"
+              when "s", "short"
+                " -s"
+              else
+                "ERROR"
+              end
+      end
+      if opt == "ERROR"
+        skip_resource "The `sys_info.hostname` resource is not supported with that option on your OS."
+      elsif opt == " -f"
+        inspec.command("hostname").stdout.chomp
+      elsif opt == " -d"
+        inspec.command("hostname").stdout.chomp.split(".", 2).last
+      elsif opt == " -s"
+        inspec.command("hostname").stdout.chomp.split(".", 2).first
+      elsif opt == " -I"
+        inspec.command("getent hosts `hostname`").stdout.chomp.split("\t").first
+      else
+        # To allow mock tests to work
+        inspec.command("hostname").stdout.chomp
+      end
+    end
     # returns the Manufacturer of the local system
```

Comment:
The linter is complaining about the lack of a blank line here - you can add it manually, or check for linter issues using `bundle exec chefstyle` . Add `-a` to have them auto-corrected.

Reply:
I'm adding it manually.  Thanks for the heads up

Reply:
What's the best way to add files or commits to PR?

Reply:
Resolved